cmd/kops/update_cluster.go

```diff
@@ -163,6 +171,7 @@ func NewCmdUpdateCluster(f *util.Factory, out io.Writer) *cobra.Command {
 	cmd.RegisterFlagCompletionFunc("lifecycle-overrides", completeLifecycleOverrides)
 
 	cmd.Flags().BoolVar(&options.Prune, "prune", options.Prune, "Delete old revisions of cloud resources that were needed during an upgrade")
+	cmd.Flags().BoolVar(&options.IgnoreVersionSkew, "ignore-version-skew", options.IgnoreVersionSkew, "Setting this to true will force updating the kubernetes version on all instance groups, regardles of which control plane version is running")
```

Comment:
given how many policies are defined here: https://kubernetes.io/releases/version-skew-policy/

and that this only violates one of them, perhaps we should make the flag more specific? `--ignore-kubelet-version-skew`

Reply:
I see your point @rifelpet, but i think instead what we could do is keep this naming and add an additional check of whether the user is trying to update more than 3 minor versions at a time, and throw an error if that's the case unless the --ignore-version-skew flag is set. That would take care of all the version skews listed there and the flag name would then make sense. WDYT?

Reply:
So we _can_ do this, using something like:

```
	// By default we enforce the version skew between control plane and worker nodes
	options.IgnoreVersionSkew = "DoNotIgnore"
	cmd.Flags().StringVar(&options.IgnoreVersionSkew, "ignore-version-skew", options.IgnoreVersionSkew, "Setting this to true will force updating the kubernetes version on all instance groups, regardless of which control plane version is running")
	cmd.Flags().Lookup("ignore-version-skew").NoOptDefVal = "IgnoreAllSkew"
```

The thing I don't like is that it won't be very discoverable, particularly once we have more options.  The help with the above looks like this:

```
      --ignore-version-skew string[="IgnoreAllSkew"]   Setting this to true will force updating the kubernetes version on all instance groups, regardless of which control plane version is running (default "DoNotIgnore")
```

So we're going to be struggling to get a clear description here.  Maybe that just doesn't matter - this is blocking the 1.31 beta release, so maybe this is good enough?

The alternative is that we have separate flags, `--ignore-kubelet-version-skew`, `--ignore-upgrade-skip-versions` and maybe that's all (but if not we just add more).  We'd have more flags, but they would be simple.

Reply:
Having separate flags makes sense to me because users may want to toggle them separately which is easier to understand as independent flags rather than trying to choose the most appropriate flag value.

Reply:
Yeah, I think it does seem simpler as I don't think we're going to end up with _too many_ of them (and if we do, we can always hide the flags and move to an aggregated flag)

Any objections to `--ignore-kubelet-version-skew` ?  I think I can push a change if we all agree cc @rsafonseca

Reply:
No concerns from me, feel free to push the change @justinsb  :)